pkg/api/v1alpha1/cluster_webhook.go

```diff
@@ -343,3 +352,47 @@ func (r *Cluster) ValidateDelete() error {
 
 	return nil
 }
+
+func validateServerVersionSkew(new, old *Cluster, log logr.Logger) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if old.IsSelfManaged() {
```

Comment:
This field is immutable for management clusters, so set it for managed clusters only.

Reply:
But why would this validation have to know about that? This validation should be independent from the other validations if possible.

If we went and make that field mutable, how would we know that we need to update this piece too?
IMO this coupling is not necessary, but I might be missing something. What's the benefit you see on having this guard?

Reply:
Yeah the only reason was to avoid running through this at all for the self managed clusters. Instead of this I can add a check to run through the logic only if the kube versions differ from each other.

Reply:
Great idea!